From 6671e352497ca4bb07a96c48e03907065ff77d8a Mon Sep 17 00:00:00 2001
From: Ryosuke Yasuoka <ryasuoka@redhat.com>
Date: Wed, 22 May 2024 00:34:42 +0900
Subject: [PATCH] nfc: nci: Fix handling of zero-length payload packets in nci_rx_work()
Git-commit: 6671e352497ca4bb07a96c48e03907065ff77d8a
Patch-mainline: v6.10-rc1
References: git-fixes

When nci_rx_work() receives a zero-length payload packet, it should not
discard the packet and exit the loop. Instead, it should continue
processing subsequent packets.

Fixes: d24b03535e5e ("nfc: nci: Fix uninit-value in nci_dev_up and nci_ntf_packet")
Signed-off-by: Ryosuke Yasuoka <ryasuoka@redhat.com>
Reviewed-by: Simon Horman <horms@kernel.org>
Reviewed-by: Krzysztof Kozlowski <krzysztof.kozlowski@linaro.org>
Link: https://lore.kernel.org/r/20240521153444.535399-1-ryasuoka@redhat.com
Signed-off-by: Paolo Abeni <pabeni@redhat.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 net/nfc/nci/core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/nfc/nci/core.c
+++ b/net/nfc/nci/core.c
@@ -1533,7 +1533,7 @@ static void nci_rx_work(struct work_stru
 
 		if (!nci_valid_size(skb)) {
 			kfree_skb(skb);
-			break;
+			continue;
 		}
 
 		/* Process frame */
